core-java-modules/core-java-streams-3/src/test/java/com/baeldung/streams/conditional/StreamForEachIfElseUnitTest.java

```diff
@@ -24,6 +27,22 @@ public final void givenIntegerStream_whenCheckingIntegerParityWithIfElse_thenEns
 
     }
 
+    @Test
+    public final void givenIntegerStream_whenCheckingIntegerParityWithCustomConsumer_thenEnsureCorrectParity() {
+        List<Integer> ints = Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 9, 10);
+
+        Consumer<Integer> integerParityAsserter = i -> {
+            if (i.intValue() % 2 == 0) {
+                Assert.assertTrue(i.intValue() + " is not even", i.intValue() % 2 == 0);
+            } else {
+                Assert.assertTrue(i.intValue() + " is not odd", i.intValue() % 2 != 0);
+            }
+        };
```

Comment:
We can do without the unboxing:
```suggestion
        Consumer<Integer> integerParityAsserter = i -> {
            if (i % 2 == 0) {
                Assert.assertTrue(i + " is not even", i % 2 == 0);
            } else {
                Assert.assertTrue(i + " is not odd", i % 2 != 0);
            }
        };
```

Reply:
Yes, the idea was to keep the same logic as the other test cases, but yeah I will remove the intValue() method

Reply:
Done